source/extensions/stat_sinks/metrics_service/grpc_metrics_service_impl.cc

```diff
@@ -16,6 +16,15 @@ namespace Extensions {
 namespace StatSinks {
 namespace MetricsService {
 
+namespace {
+// TODO(#4160): use TimeSource::systemTime() rather than directly reading the system-clock.
```

Comment:
@jmarantz what's stopping us switching to this right away? Seems best to avoid exception cases like this if we can given how regular time is now treated in Envoy.

Reply:
nothing, now that you merged the TimeSystem PR; this TODO is addressable.

Reply:
@zyfjeff post https://github.com/envoyproxy/envoy/pull/4257 merge, do you think we can now use this time source?

Reply:
FYI the path to doing this is to plumb Event::TimeSystem& into constructors as needed, up to MainCommon. Note that both ServerImpl and Dispatcher support timeSystem() or timeSource() APIs, so once you go up the constructor stack till you have one of those, you can stop.

Once we have all the std::chrono removed from everywhere in Envoy we could decide to switch to absl::Time, though I'd want to socialize the benefits of that to envoy-dev on slack first.

Reply:
@htuch Yes i do, I am going to modify it.

@jmarantz  I can try to replace all std::chrono with TimeSystem